awx/main/scheduler/task_manager.py

```diff
@@ -45,6 +45,12 @@ class TaskManager():
 
     def __init__(self):
         self.graph = dict()
+        # start task limit indicates how many pending jobs can be started on this
+        # .schedule() run. Starting jobs is expensive, and there is code in place to reap
+        # the task manager after 5 minutes. At scale, the task manager can easily take more than
+        # 5 minutes to start pending jobs. If this limit is reached, pending jobs
+        # will no longer be started and will be started on the next task manager cycle.
+        self.start_task_limit = 150
```

Comment:
Curious, why 150, and not - say - 100?

Reply:
No particular reason -- the original issue involved around 300 jobs locking up the task manager, so I figured half of that would be a safe number.

Reply:
I could see a benefit to keeping the definition of this constant in settings.

Reply:
i'll look into adding this as a setting